docs/devguide/hexformat.rst

```diff
@@ -67,3 +105,48 @@ Running the ``make all`` command executes the following steps:
 - The ``tools/adduicr.py`` script adds the UICR to the bare hex
 - The final hex file is placed in ``build/firmware.hex``
 - The user can optionally append a script using ``tools/makecombinedhex.py`` (or other tools)
+
+micro:bit V2
+============
+
+This applies to MicroPython for the micro:bit V2, the source of which can be found here: `microbit-foundation/micropython-microbit-v2 <https://github.com/microbit-foundation/micropython-microbit-v2>`_.
+
+This is a port of MicroPython to the micro:bit which uses the CODAL as the underlying target platform.
+
+After cloning this repository update the submodules:
+
+`$ git submodule update --init`
+
+Then build the MicroPython cross-compiler:
+
+`$ make -C lib/micropython/mpy-cross`
+
+After setting up, go to the src/ directory and build:
+
+`$ cd src`
+
+`$ make`
+
+That will build both libmicropython.a (from source in src/codal_port/) and the CODAL app (from source in src/codal_app/). The resulting firmware will be MICROBIT.hex in the src/ directory which can be copied to the micro:bit.
+
+Including a user script
+-----------------------
+
+User scripts are stored in the MicroPython filesystem and if a `main.py` script exists it is run when MicroPython starts. Additional Python scripts can also be included and executed from the main.py file, or the REPL.
```

Comment:
Only because previous subsection separated V1/V2 description, it might be useful to include here that this method is for both version of MicroPython.